pkg/targetgroup/deployment.go

```diff
@@ -49,25 +49,27 @@ type Config struct {
 	Type  string      `json:"type" dynamodbav:"type"`
 	Value interface{} `json:"value" dynamodbav:"value"`
 }
+
 type Provider struct {
 	Publisher string `json:"publisher" dynamodbav:"publisher"`
 	Name      string `json:"name" dynamodbav:"name"`
 	Version   string `json:"version" dynamodbav:"version"`
 }
+type ConfigValidation struct {
+	Logs    []Diagnostic `json:"logs"`
+	Success bool         `json:"success"`
+}
 
 type ProviderDescribe struct {
-	Provider         Provider          `json:"provider"`
-	Config           map[string]string `json:"config"`
-	ConfigValidation map[string]struct {
-		Logs    []Diagnostic `json:"logs"`
-		Success bool         `json:"success"`
-	} `json:"configValidation"`
-	Schema struct {
+	Provider         Provider                    `json:"provider"`
+	Config           map[string]Config           `json:"config"`
```

Comment:
```suggestion
	Config           map[string]string          `json:"config"`
```
Config returned by ProviderDescribe will only have config value and won't have a type. Is the type even required here?

Reply:
@Eddie023 In this case, I want to update the provider framework to return this custom type rather than just a string